### ISSUE TITLE
Write a user-friendly README targeted at external developers

### ISSUE THREAD

**Issue by PaddyMc** [strong-config/node#14]
## tl;dr
We need a README that makes it as easy and quick as possible for potential strong-config users to get started using strong-config 💪

## Recommended Reading
https://www.divio.com/blog/documentation/

IMO what we're trying to achieve with the README in the scope of this article is a "How-To Guide". There can be links into other forms of documentation in the README such as "Explanation", "Reference", or "Tutorial" but the README itself should stick to a high-level overview that explains how to solve the most important things quickly and easily.

## Target Audience: New Users
This is our primary target audience for the README. Nobody has ever heard about strong-config outside of our walls, so we should make sure our README is written in a way that addresses new users and does not require _any_ previous knowledge or Brickblock-internal context to get going.

## Content
1. **Intro**
   Super brief intro on what strong-config is, what problem it solves, and what people can use it for (e.g. "Replace `dotenv`"). Mention Brickblock.

1. **tl;dr**
   A super condensed section for people that just wanna get going right away. Something like:
   - Clone the repo: `git clone ...`
   - Check out `example/schema.json` and `example/development.yml` and make some changes
   - Run `yarn dev:load` to see the config object that would get loaded in your application

1. **How to use**
   - Install package: `yarn add @strong-config/node`
   - Define config files: `touch config/development.yml config/production.yml` then give example of a simply config file...
   - Load config: Add `export const config = new StrongConfig().load()` to the entrypoint of your application (e.g. your `./src/index.js`
   - Use config: In all other files, simply `import { config } from '../path/to/entrypoint'` and `console.log(config.logger.level)`
   - Options-section describing possible configuration options

1. **How to encrypt config values**
   - Set up KMS provider
     - Give examples for Google KMS / AWS KMS / Azure KMS (maybe just link to individual markdown docs that each have a full walkthrough for setting it up on the respective cloud provider incl. permissions etc.)
   - Add `Secret` suffix, for example `apiKeySecret`
   - `yarn strong-config encrypt production`
   - Check `production.yml`

1. **How to decrypt config values**
   - `yarn strong-config decrypt production`

1. **FAQ**
   - Gather all commonly asked questions incl. an answer here, such as:
     - Is it safe to check in my secrets?
     - etc

1. **Contributing**
   Explain how to contribute to this repo, where would we most like help? How do we want people to help? What's our code of conduct? How to develop locally? How to test? etc.

**Comment by crosby** [comment]
Should we move the contributing section to a `CONTRIBUTING.md` file?

**Comment by chapati23** [comment]
Yeah, think that's also quite common in other projects. Let's do it 👍

**Comment by mohoff** [comment]
Few comments in the light of having thought about this while working on #29 

- step 2.: I'd skip it. `git clone` and `dev:scripts` are more steps for developers then for people wanting to get started using our tool.
- step 4.: Depends on key provider setup, and `decrypt/encrypt` commands. Also the given command is probably part of `strong-config/cli`. IMO the line between the `node` and `cli` part of strong-config is narrow, but for now we should focus the README on the `node` part.
- step 5.: same here, this is more `strong-config/cli` syntax.

I like step 1,3,6,7 as outlined :)
